src/classic/class/ReactClass.js

```diff
@@ -776,10 +765,10 @@ var ReactClassMixin = {
    * @deprecated
    */
   replaceProps: function(newProps, callback) {
-    ReactInstanceMap.get(this).replaceProps(
-      newProps,
-      callback
-    );
+    ReactUpdateQueue.enqueueReplaceProps(this, newProps);
+    if (callback) {
+      ReactUpdateQueue.enqueueCallback(callback);
```

Comment:
Also, would it make sense to just pass the callback through to `enqueueReplaceProps` (and family) as the last arg. We're just always checking for it in all of these cases anyway.